fw/sock.c

```diff
@@ -751,6 +751,12 @@ ss_tcp_process_skb(struct sock *sk, struct sk_buff *skb, int *processed)
 		WARN_ON_ONCE(skb_has_frag_list(skb));
 		WARN_ON_ONCE(skb->sk);
 
+		/*
+		 * Some SKBs may have dev, however tempesta uses dev to store
+		 * own flags, thus clear it.
+		 */
+		skb->dev = NULL;
```

Comment:
How this initialization correlates with the `skb_tfw_is_present()` check in `ss_skb_init_for_xmit()`. It seems we nullify `dev` for all ingress `skb`s, so does `ss_skb_init_for_xmit()` ever sees non-NULL `dev`?

Also why do we need to initialize `dev` here? Our code doesn't use except the case to store the flags on transmission path and this case is handled with `ss_skb_init_for_xmit()`.